libs/ngxtension/svg-sprite/src/svg-sprite.ts

```diff
@@ -0,0 +1,471 @@
+import {
+	Directive,
+	ENVIRONMENT_INITIALIZER,
+	ElementRef,
+	Injectable,
+	Injector,
+	Input,
+	NgZone,
+	booleanAttribute,
+	computed,
+	effect,
+	inject,
+	makeEnvironmentProviders,
+	runInInjectionContext,
+	signal,
+	type OnInit,
+} from '@angular/core';
+import { toObservable, toSignal } from '@angular/core/rxjs-interop';
+import { filterNil } from 'ngxtension/filter-nil';
+import { from, map, shareReplay, switchMap, type Observable } from 'rxjs';
+
+/**
+ * Represents a svg sprite.
+ *
+ * @see {@link NgxSvgSpriteFragment.sprite}
+ * @see {@link NgxSvgSprites.register}
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use `use` Element}
+ */
+export interface NgxSvgSprite {
+	/**
+	 * A unique name identifying this sprite.
+	 */
+	name: string;
+
+	/**
+	 * E.g. `path/to/my/sprite.svg`
+	 */
+	baseUrl: string;
+
+	/**
+	 * @param baseUrl reference to this {@link baseUrl}
+	 * @param fragment
+	 * @returns a url pointing to a specified svg in this sprite by using a `fragment` e.g `path/to/my/sprite.svg#${fragment}`
+	 *
+	 * @see {@link https://svgwg.org/svg2-draft/linking.html#URLReference}
+	 */
+	url: (baseUrl: string, fragment: string) => string;
+
+	/**
+	 * Whether to copy the `viewBox` attribute from the `symbol` in the svg sprite.
+	 */
+	autoViewBox?: boolean;
+
+	/**
+	 *
+	 * @param fragment
+	 * @returns a list of classes that are applied to the svg element.
+	 */
+	classes?: (fragment: string) => string[] | string;
+}
+
+export type CreateSvgSpriteOptions = Omit<NgxSvgSprite, 'url'> &
+	Partial<Pick<NgxSvgSprite, 'url'>>;
+
+/**
+ * Creates a {@link NgxSvgSprite}.
+ *
+ * @param options
+ * @returns
+ */
+export const createSvgSprite = (options: CreateSvgSpriteOptions) => {
+	if (options.url == null)
+		options.url = (baseUrl, fragment) => `${baseUrl}#${fragment}`;
+
+	return options as NgxSvgSprite;
+};
+
+/**
+ * This service registers {@link NgxSvgSprite svg sprites}, which can be rendered via {@link NgxSvgSpriteFragment}.
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxSvgSprites {
+	/**
+	 * @ignore
+	 */
+	private readonly ngZone = inject(NgZone);
+
+	/**
+	 * <{@link NgxSvgSprite.name}, {@link NgxSvgSprite}>
+	 */
+	private readonly sprites: Record<
+		string,
+		NgxSvgSprite & { svg$: Observable<SVGGraphicsElement> }
+	> = {};
+
+	/**
+	 * Registers a sprite.
+	 *
+	 * @param sprite
+	 *
+	 * @see {@link NgxSvgSpriteFragment.sprite}
+	 */
+	public readonly register = (sprite: NgxSvgSprite) => {
+		this.sprites[sprite.name] = {
+			...sprite,
+			svg$: from(
+				this.ngZone.runOutsideAngular(() => fetch(sprite.baseUrl)),
+			).pipe(
+				switchMap((response) => response.text()),
+				map((text) => {
+					const svg = new DOMParser()
```

Comment:
nit/question: I'm not too familiar with `DOMParser`. Can we just instantiate `DOMParser` once instead?

Reply:
I refactored this part from fetch to xhr, because I found out xhr can parse documents. DOMParser is not needed with xhr.